static/app/views/settings/projectSourceMaps/projectSourceMapsArtifacts.tsx

```diff
@@ -230,7 +232,7 @@ export function ProjectSourceMapsArtifacts({params, location, router, project}:
           ? debugIdBundlesData?.[0].map(data => {
               const downloadUrl = `${api.baseUrl}/projects/${organization.slug}/${
                 project.slug
-              }/releases/${encodeURIComponent(data.debugId)}/files/${
+              }/releases/${encodeURIComponent(params.bundleId)}/files/${
```

Comment:
one of the nice things about testing... I was able to catch this bug. The URL was wrong and this change shall fix it :)